allauth_2fa/views.py

```diff
@@ -66,20 +66,16 @@ class TwoFactorSetup(FormView):
     success_url = reverse_lazy('two-factor-backup-tokens')
 
     def dispatch(self, request, *args, **kwargs):
+        if request.user.totpdevice_set.filter(confirmed=True).exists():
+            return HttpResponseRedirect(reverse_lazy('two-factor-backup-tokens'))
 
-        if 'allauth_otp_qr_secret_key' not in request.session:
-            self.secret_key = random_hex(20).decode('ascii')
-            request.session['allauth_otp_qr_secret_key'] = self.secret_key
-        else:
-            self.secret_key = request.session['allauth_otp_qr_secret_key']
+        request.user.totpdevice_set.filter(confirmed=False).delete()
+        TOTPDevice(user=request.user, confirmed=False).save()
 
-        if request.user.totpdevice_set.exists():
-            return HttpResponseRedirect(reverse_lazy('two-factor-backup-tokens'))
         return super(TwoFactorSetup, self).dispatch(request, *args, **kwargs)
 
-    def get_form_kwargs(self):
+    def get_form_kwargs(self, request):
```

Comment:
This change seems wrong, [`get_form_kwargs`](https://docs.djangoproject.com/en/1.10/ref/class-based-views/mixins-editing/#django.views.generic.edit.FormMixin.get_form_kwargs) doesn't take any parameters.